src/Database/LSMTree/Internal/Serialise/RawBytes.hs

```diff
@@ -211,10 +229,10 @@ fromByteArray :: Int -> Int -> ByteArray -> RawBytes
 fromByteArray off len ba = RawBytes (P.Vector off len ba)
 
 pack :: [Word8] -> RawBytes
-pack = coerce P.fromList
+pack = RawBytes . P.fromList
```

Comment:
e.g. here we "unnecessarily" double-check that `P.fromList` doesn't violate `Vector`s invariants. This doesn't feel right to be so overly paranoid.